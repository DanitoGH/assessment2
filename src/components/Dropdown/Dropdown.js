import React from 'react';
import PropTypes from 'prop-types';

import { FormControl } from "react-bootstrap";

const Dropdown = ({options,dropdownCategory, roomChecked, roomNumber,adultRoomValue, childrenRoomValue, onChange }) => {

    const handleChange = (e) => {
        const { value } = e.target;
        onChange({
            Category: dropdownCategory,
            selectedValue: value,
            roomNumber: roomNumber 
        })
    }

    const defaultValue = (dropdownCategory === 'Adults'? 1 : 0)
    const defaultSelectedValue = (dropdownCategory === 'Adults'? adultRoomValue: childrenRoomValue)

    return (
        <FormControl
                className="mt-1 mb-2"
                as="select"
                disabled={ !roomChecked? true : false }
                value={ adultRoomValue && childrenRoomValue !== ''? defaultSelectedValue : defaultValue  }
                onChange={handleChange}>
                { !roomChecked? dropdownCategory === 'Adults'? 
                    <option value={1}>1</option> : 
                    <option value={0}>0</option> : null 
                }
                {
                roomChecked && options.map((option, key) => (
                    <option 
                        key={key}
                        value={option.value}
                    >
                    { option.label}
                    </option>
                ))
                }
         </FormControl>
    )
}

Dropdown.protoTypes = {
    options: PropTypes.array.isRequired,
    roomChecked: PropTypes.bool.isRequired,
    dropdownCategory: PropTypes.string.isRequired
}
 
 export default Dropdown;