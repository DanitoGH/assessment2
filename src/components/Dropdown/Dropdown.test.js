import { render } from '@testing-library/react';
import { shallow }  from "enzyme";
import toJSON from "enzyme-to-json";

import Dropdown from './Dropdown';


test('properly renders dropdown component', () => {
  render(<Dropdown />);
})

test("matches snapshot", () => {
  const wrapper = shallow(<Dropdown />);

  expect(toJSON(wrapper)).toMatchSnapshot();
})
