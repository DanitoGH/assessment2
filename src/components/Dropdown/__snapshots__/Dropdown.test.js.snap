// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<FormControl
  as="select"
  className="mt-1 mb-2"
  disabled={true}
  onChange={[Function]}
  value={0}
>
  <option
    value={0}
  >
    0
  </option>
</FormControl>
`;
